server/src/main/java/org/opensearch/index/translog/transfer/TranslogTransferManager.java

```diff
@@ -296,4 +296,52 @@ public void onFailure(Exception e) {
             }
         );
     }
+
+    public void deleteStaleTranslogMetadataFilesAsync(int lastNMetadataFilesToKeep) {
+        transferService.listAllAsync(ThreadPool.Names.REMOTE_PURGE, remoteMetadataTransferPath, new ActionListener<>() {
```

Comment:
This will be called more frequently than `deleteRemoteGenerationAsync`, right? And it will delete only one metadata file in each call. Thinking if we can club these calls to reduce the frequency.

Reply:
https://github.com/opensearch-project/OpenSearch/pull/6238#issuecomment-1423766643

Reply:
Incorporated.